draft-ietf-quic-transport.md

```diff
@@ -678,6 +687,34 @@ packet protection in detail.  After decryption, the plaintext consists of a
 sequence of frames, as described in {{frames}}.
 
 
+## Compound Packets {#packet-compound}
+
+Compound packets allow the sender to assemble an Initial ({{packet-initial}})
+or a Handshake ({{packet-handshake}}) packet and a Protected
+({{packet-protected}}) packet into one UDP packet, thereby reducing the number
+of packets required to be emitted when application data can be sent during the
+handshake.
+
+A Compound-Initial packet is identical to an Initial packet with the exception
+being that it has the Payload Length field.  The field designates the length of
+the payload of the Initial packet.  The remainder of the UDP packet contains a
+0-RTT Protected packet.
+
+A Compound-Handshake packet is identical to a Handshake packet with the
+exception being that it has the Payload Length field.  The field designates the
+length of the payload of the Handshake packet.  The remainder of the UDP packet
+contains either a 0-RTT Protected packet or a short header packet
+({{short-header}}).
+
+The sender MUST NOT assemble QUIC packets belonging to different QUIC
+connections into a single Compound packet.
+
+The receiver of a Compound packet MUST individually process the Cryptographic
+Handshake packet and the Protected packet being contained.  It MUST process the
+Cryptographic Handshake packet of a Compound packet even when the Protected
+packet being contained is deemed invalid.
```

Comment:
I think I'd simplify away these terms for the sections of the packet.  Simply say that the thing which follows is processed as a separate packet, which might be discarded, buffered, or processed depending on the receiver's current state.

Reply:
...and contains its own packet number, and therefore gets acknowledged separately.